From 940acabcc08b73c82d4fe0f9827ab54ab1ece282 Mon Sep 17 00:00:00 2001
From: Manjunath <manjunathx.b.s@intel.com>
Date: Mon, 10 Sep 2012 12:57:36 +0530
Subject: [PORT FROM MAIN] Video with WMA9-Pro audio can't be played

BZ: 56385

The AsfExtractor is changed to recognise WMA_Pro audio
content file to have mimetype as "audio/unknown-type".
In AwesomePlayer.cpp, the mimetype is checked and
setAudioSource is not done for WMA_Pro file.

Change-Id: I88cdeb3ef88223d17a85e9719be17be1aaaf55a5
Orig-Change-Id: I158a84f64ae632bd5e8e8657e132aed37cecbfe1
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>
Reviewed-on: http://android.intel.com:8080/80772
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index 984095f..e216545 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -426,6 +426,8 @@ status_t AwesomePlayer::setDataSource_l(const sp<MediaExtractor> &extractor) {
                     &mStats.mTracks.editItemAt(mStats.mVideoTrackIndex);
                 stat->mMIME = mime.string();
             }
+        } else if (!haveAudio && !strncasecmp(mime.string(), "audio/unknown-type",18 )) {
+            haveAudio = false;
         } else if (!haveAudio && !strncasecmp(mime.string(), "audio/", 6)) {
             setAudioSource(extractor->getTrack(i));
             haveAudio = true;
-- 
1.7.4.1

